### ISSUE TITLE
Drop the WORKSPACE API for apt.

### ISSUE THREAD

**Issue by thesayyn** [GoogleContainerTools/rules_distroless#125]
Currently, none of our interested parties are using the WORKSPACE api of the apt, therefore its just a maintenance burden on us going forward. 

With https://github.com/GoogleContainerTools/distroless/pull/1640 we moved distroless repo to use BZLMOD.

**Comment by jjmaestro** [comment]
YASSSS! I was really looking forward to this, it's going to help a lot removing a bunch of stuff in the codebase and enabling much better architecture (what you mentioned in #123).

Let me know if you want me to make a PR, happy to give it a go!

**Comment by alexeagle** [comment]
I don't think we should drop the WORKSPACE API while Bazel 7 is still in LTS, will be a while before all users are on bzlmod still.
